From dc0dba39a90b0ffe63bf63eec45b3e338f3b485c Mon Sep 17 00:00:00 2001
From: Ding Yunliang <yunliang.ding@intel.com>
Date: Fri, 10 Apr 2015 11:37:48 +0800
Subject: [PATCH 349/351] atomisp: no stop SP if start err

There is some unknown reason when starting SP timeout, in this
case, stop SP should not be called.

Change-Id: I44b0278d56da39d55361c6e80ab92423a36182b7
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-27935
Signed-off-by: Ding Yunliang <yunliang.ding@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/50628
Approver: Bin B Han <bin.b.han@intel.com>
Reviewed-by: Bin B Han <bin.b.han@intel.com>
Validation-Android: Wei Tao <wei.tao@intel.com>
Integrator: Wei Tao <wei.tao@intel.com>
Build: Wei Tao <wei.tao@intel.com>
Reviewed-by: Wei Tao <wei.tao@intel.com>
Maintainer: Wei Tao <wei.tao@intel.com>
---
 .../drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c
index 2040b8e..49882a0 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c
@@ -1260,6 +1260,7 @@ int atomisp_css_start(struct atomisp_sub_device *asd,
 			enum atomisp_css_pipe_id pipe_id, bool in_reset)
 {
 	struct atomisp_device *isp = asd->isp;
+	bool sp_is_started = false;
 	int ret = 0, i = 0;
 	if (in_reset) {
 		if (__destroy_streams(asd, true))
@@ -1318,6 +1319,8 @@ int atomisp_css_start(struct atomisp_sub_device *asd,
 			dev_err(isp->dev, "start sp error.\n");
 			ret = -EINVAL;
 			goto start_err;
+		} else {
+			sp_is_started = true;
 		}
 	}
 
@@ -1348,7 +1351,7 @@ stream_err:
 	/*
 	 * SP can not be stop if other streams are in use
 	 */
-	if (atomisp_streaming_count(isp) == 0)
+	if ((atomisp_streaming_count(isp) == 0) && sp_is_started)
 		ia_css_stop_sp();
 
 	return ret;
-- 
1.9.1

